apptest: improve routine that download tflite files @open sesame 03/27 17:58

leemgs opened the pull request:
Fixed issue https://github.com/nnsuite/nnstreamer/issues/1265.

The existing apptest module always download the same files such as
a network model and label file repeatedly. This commit is to improve
the existing operation to reduce a waiting time and save a network
cost of AWS instance.

Signed-off-by: Geunsik Lim <geunsik.lim@samsung.com>


---

leemgs commented:
Starting an aging test.

jaeyun-jung commented:
[suggestion]
Currently app-test module uses only one model/label. 
I'm not sure that we may add another nn model for app test later. For this case, I suggest to make directory /tmp/nn_model/tflite_model and copy downloaded model.

leemgs commented:
**Changelog**

    * Version 2:
    1. Added /tmp/nnsuite/ folder for more readability and maintenance
       - /tmp/nnsuite/tflite_model/ foldr in case of TF-lite model

    * Version 1:
    1. Added a routine that archive the network model files to save a network cost
       - AWS instance reboot at 5:00 AM everyday
       - apptest download the network files at once
       - Then apptest does not downoad the files repeatedly by using /tmp/nnsuite/* files.

leemgs commented:
The aging test is successfully done. This PR is ready to be merged.

leemgs commented:
> e.g., get_file_cached (URL, LINKPATH, FULLCACHEPATH)

Applied.

leemgs commented:
**Changelog**
* V3
1. Added func_get_file_cached() to avoid duplicated statements.
2. Updated annotations

* V2
1. Added /tmp/nnsuite/ folder for more readability and maintenance
   - /tmp/nnsuite/tflite_model/ foldr in case of TF-lite model

* V1
1. Added a routine that archive the network model files to save a network cost
   - AWS instance reboot at 5:00 AM everyday
   - apptest download the network files at once
   - Then apptest does not downoad the files repeatedly by using /tmp/nnsuite/* files

leemgs commented:
Starting an aging test.

leemgs commented:
The aging test is successfully completed. The test is executed with https://github.com/nnsuite/nnstreamer/pull/1226.

leemgs commented:
This PR is ready to be merged.

jaeyun-jung commented on ci/taos/plugins-staging/pr-audit-nnstreamer-ubuntu-apptest.sh:
```diff
@@ -143,19 +169,37 @@ function pr-audit-nnstreamer-ubuntu-apptest-run-queue() {
-    wget -a wget.log https://github.com/nnsuite/testcases/raw/master/DeepLearningModels/tensorflow-lite/Mobilenet_v1_1.0_224_quant/labels.txt
-    result+=$?
-    cd ..
+    file_label="labels.txt"
```
I think this line is unnecessary

leemgs replied:
Nice catch. :)

leemgs replied:
Updated.
